src/backend/routers/scim.py

```diff
@@ -0,0 +1,259 @@
+from typing import Optional
+
+from fastapi import APIRouter
+from starlette.requests import Request
+from starlette.responses import JSONResponse, Response
+
+import backend.crud.group as group_crud
+import backend.crud.user as user_crud
+from backend.config import Settings
+from backend.config.routers import RouterName
+from backend.database_models import DBSessionDep
+from backend.database_models import Group as DBGroup
+from backend.database_models import User as DBUser
+from backend.database_models import UserGroupAssociation
+from backend.schemas.scim import (
+    CreateGroup,
+    CreateUser,
+    Group,
+    ListGroupResponse,
+    ListUserResponse,
+    PatchGroup,
+    PatchUser,
+    UpdateUser,
+    User,
+)
+
+SCIM_PREFIX = "/scim/v2"
+scim_auth = Settings().auth.scim
+router = APIRouter(prefix=SCIM_PREFIX)
+router.name = RouterName.SCIM
+
+
+class SCIMException(Exception):
+    def __init__(self, status_code: int, detail: str):
+        self.status_code = status_code
+        self.detail = detail
+
+
+async def scim_exception_handler(request: Request, exc: SCIMException) -> Response:
+    return JSONResponse(
+        status_code=exc.status_code,
+        content={
+            "detail": exc.detail,
+            "schemas": ["urn:ietf:params:scim:api:messages:2.0:Error"],
+        },
+    )
+
+
+@router.get("/Users")
+async def get_users(
+    session: DBSessionDep,
+    count: int = 100,
+    start_index: int = 1,
```

Comment:
Why 1 not 0?

Reply:
This is in the spec 🤷

Reply:
https://datatracker.ietf.org/doc/html/rfc7644
startIndex  The 1-based index of the first result in the current set
of list results.  REQUIRED when partial results are returned due
to pagination.